File: 20180711/Move Zeros.py
class Solution(object):
    def exchange(self, a, b):
        return b,a
    def moveZeroes(self, nums):
        """
        :type nums: List[int]
        :rtype: void Do not return anything, modify nums in-place instead.
        """
        i = -1
        j = i + 1
        p = len(nums) - 1
        while j <= p:
            if nums[j] != 0:
                i += 1
                nums[i], nums[j] = self.exchange(nums[i], nums[j])
            j += 1
        

File: 20180711/Move Zeros.java
class Solution {
    public void moveZeroes(int[] nums) {
        int i = -1;
        int j = i + 1;
        int p = nums.length - 1;
        while (j <= p){
            if(nums[j] != 0){
                i += 1;
                int tmp = nums[i];
                nums[i]  = nums[j];
                nums[j] = tmp;
            }
            j += 1;
        }
    }
}
